tests/pocket_ic_test.py

```diff
@@ -8,78 +8,106 @@
 
 # The test needs to have the module in its sys path, so we traverse
 # up until we find the pocket_ic package.
-
 sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
-from pocket_ic import PocketIC
+from pocket_ic import PocketIC, SubnetKind, NNS, STANDARD
 
 
 class PocketICTests(unittest.TestCase):
-    def setUp(self) -> None:
-        # This is being run for every test independently.
-        self.pic = PocketIC()
-        return super().setUp()
-
-    def tearDown(self) -> None:
-        # Delete the current PocketIC instance after the test has executed.
-        self.pic.delete()
-        return super().tearDown()
-    
+    def test_multiple_nns_subnets_are_deduplicated(self):
+        subnet_config = [NNS, NNS]
+        pic = PocketIC(subnet_config)
+        self.assertEqual(len(pic.topology), 1)
+
+    def test_install_canister_on_subnet_and_get_subnet_of_canister(self):
+        subnet_config = [NNS, STANDARD]
+        pic = PocketIC(subnet_config)
+        nns_subnet = next(
+            k for k, v in pic.topology.items() if v.subnet_kind == SubnetKind.NNS
+        )
+        app_subnet = next(
+            k
+            for k, v in pic.topology.items()
+            if v.subnet_kind == SubnetKind.APPLICATION
+        )
+        nns_canister = pic.create_canister(subnet=nns_subnet)
+        app_canister = pic.create_canister(subnet=app_subnet)
+
+        self.assertEqual(
+            pic.get_subnet_of_canister(nns_canister).bytes, nns_subnet.bytes
+        )
+        self.assertEqual(
+            pic.get_subnet_of_canister(app_canister).bytes, app_subnet.bytes
+        )
+
     def test_set_get_stable_memory_no_compression(self):
-        canister_id = self.pic.create_canister()
-        self.pic.install_code(canister_id, b"\x00\x61\x73\x6d\x01\x00\x00\x00", [])
+        pic = PocketIC()
+        canister_id = pic.create_canister()
+        pic.install_code(canister_id, b"\x00\x61\x73\x6d\x01\x00\x00\x00", [])
```

Comment:
Where does this magic bytestring come from?

Reply:
That's a minimal correct canister wasm. I used it because we don't need a specific canister here